### PR TITLE
Rewrite Tile as Fn compoanent

### PR DESCRIPTION
**Pull request by cquiroz**
Another rewrite, Took me a bit to understand how to properly use a Ref in a hook

### CODE REVIEW

File: common/src/main/scala/explore/components/Tile.scala [hunk 1]
```diff
@@ -38,13 +37,14 @@ final case class Tile(
...
 )(val render:        Tile.RenderInTitle ==> VdomNode)
-    extends ReactProps[Tile](Tile.component) {
+    extends ReactFnProps[Tile](Tile.component) {
   def showMaximize: Boolean                                                                =
     state === TileSizeState.Minimized || (canMaximize && state === TileSizeState.Normal)
   def showMinimize: Boolean                                                                =
     state === TileSizeState.Maximized || (canMinimize && state === TileSizeState.Normal)
   def withState(state: TileSizeState, sizeStateCallback: TileSizeState ==> Callback): Tile =
     copy(state = state, sizeStateCallback = sizeStateCallback)(render)
+  protected val infoRef                                                                    = Ref.toVdom[html.Element]
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:47]: Is this used?

File: common/src/main/scala/explore/components/Tile.scala [hunk 2]
```diff
@@ -71,89 +67,72 @@ object Tile {
...
-      <.div(ExploreStyles.Tile |+| ExploreStyles.FadeIn, p.key.whenDefined(^.key := _))(
-        <.div(
-          ExploreStyles.TileTitle,
-          p.back.map(b => <.div(ExploreStyles.TileButton, b)),
-          Menu(
+  val component =
+    ScalaFnComponent
+      .withHooks[Props]
+      // info ref
+      .useRefToVdom[html.Element]
```
Comment (rpiaggio) [inline common/src/main/scala/explore/components/Tile.scala:74]: Cool, I didn't know this existed.

**Review by rpiaggio — APPROVED** [review]
LGTM

File: common/src/main/scala/explore/components/Tile.scala (hunk 1 above)
Comment (cquiroz) [reply common/src/main/scala/explore/components/Tile.scala:47]: Nope, I'll remove it

File: common/src/main/scala/explore/components/Tile.scala [hunk 3]
```diff
@@ -71,89 +67,72 @@ object Tile {
...
+      .render { (p, infoRef) =>
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/Tile.scala:75]: I tried to do `renderWithReuse` and I'm facing troubles to define `Reusability` for the ref

Comment (rpiaggio) [reply common/src/main/scala/explore/components/Tile.scala:75]: A ref is a mutable holder (like cats-effect `Ref`). The ref is always the same, even though its `current` value may change. So I think it's safe to use `Reusability.always`.

File: common/src/main/scala/explore/components/Tile.scala [hunk 4]
```diff
@@ -53,13 +56,11 @@ object Tile {
...
-  protected case class State(portalNode: Option[Raw.ReactDOM.Container] = None)
-
   implicit val propsReuse: Reusability[Tile] = Reusability.derive && Reusability.by(_.render)
 
-  implicit val rawContainerReuse: Reusability[Raw.ReactDOM.Container] = Reusability.always
-
-  implicit val stateReuse: Reusability[State] = Reusability.derive
+  implicit val rawContainerReuse: Reusability[html.Element]                  = Reusability.always
+  implicit val rawDomContainerReuse: Reusability[Raw.ReactDOM.Container]     = Reusability.always
+  implicit def ref[F[_]]: Reusability[Ref.FullF[F, Node, Node, HTMLElement]] = Reusability.always
```
Comment (cquiroz) [inline common/src/main/scala/explore/components/Tile.scala:63]: Kind of ugly

**Pull request merged**